File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        
        int area=0;
        int low=0;
        int high=heights.length-1;
        int currentArea=maxAr(heights,low,high);
        while(low<high){
            if(heights[low]<heights[high]){
                
                area=maxAr(heights,low,high);
                if(area>currentArea){
                    currentArea=area;
                }
                low++;
            // }if(heights[low]<heights[high-1]){
            //     high--;
            //     area=maxAr(heights,low,high);
            //     if(area>currentArea){
            //         currentArea=area;
            //     }
            }
            else{
               
                area=maxAr(heights,low,high);
                if(area>currentArea){
                    currentArea=area;
                }
                 high--;
            }
          
        }
        return currentArea;
    }

    int maxAr(int[] heights,int low,int high){
        int hight=0;
        int base=high-low;
        if(heights[high]>heights[low]){
            hight=heights[low];
        }else{
            hight=heights[high];
        }
        return base*hight;
       

    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        low = 0
        high = len(s) -1

        def isPalindrome(s , l , h):
            while(l <= h):
                if (s[l] != s[h]):
                    return False
                l += 1
                h -= 1
            return True

        while(low <= high):
            if(s[low] != s[high]):
                return (isPalindrome(s , low +1 ,high) or isPalindrome(s, low ,high-1))
            
            low += 1
            high -= 1

        return True


        
            


            
        